### PR TITLE
SourceBuffer.mode with generate timestamps flag true

### PR DESCRIPTION
**Pull request by plehegar**

### CODE REVIEW

**Comment by gsnedders** [comment]
```
15:01 < aryehgregor> Um, I might have done something wrong by mistake.
15:01 < aryehgregor> I think I pushed to the wrong repo?
```

File: media-source/mediasource-sourcebuffer-mode-timestamps.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                          function() {
+                              sourceBuffer.mode = "segments";
+                          },
+                          'SourceBuffer#mode with generate timestamps flag true');
+        }), false);
+        var video = document.createElement('video');
+        video.src = window.URL.createObjectURL(mediaSource);
+    }, mime + ' : ' +
+       'If generate timestamps flag equals true and new mode equals "segments", ' +
+       'then throw an INVALID_STATE_ERR exception and abort these steps.');
```
Comment (wolenetz) [inline media-source/mediasource-sourcebuffer-mode-timestamps.html]: Here, and in the assert_throws, above, this should now be a TypeError (see https://github.com/w3c/media-source/pull/47).

**Comment by wolenetz** [comment]
I'll fix the assert_throws to check for TypeError and merge this shortly. Note that Chromium fails this since https://crbug.com/607372 needs fixing still. Underneath the hood, Chromium forces sequence mode behavior for these generate-timestamp-flag==true streams, but doesn't expose 'sequence' as their mode via the attribute, nor does it disallow setting that mode to 'segments'.

**Comment by wolenetz** [comment]
LGTM with my update in af85af6. I'll merge this shortly.

**Pull request merged**